### PR TITLE
Move restrictions

### PR DESCRIPTION
**Pull request by valeriegc**
Cleaned up restrictions for moves : please note : does not include the scenarios with the opponent, but rather just basic movement rules for each of the pieces.

### CODE REVIEW

File: src/functions/moveChecks/queenCheck.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { biggestSquare, columnFinder, firstColumn, firstRow, lastColumn, lastRow, rowFinder, smallestSquare, diagonalRowNE, diagonalRowNW, diagonalRowSE, diagonalRowSW } from "../../global"
+import { allowedSquares } from "../../stores"
+
+export const queenCheck = (targetSquare:number) => {
```
Comment (AlexanderKolberg) [inline src/functions/moveChecks/queenCheck.ts:4]: Could this be replaced by using the tower and bishop?

File: src/functions/moveChecks/bishopCheck.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { allowedSquares } from "../../stores"
+
+export const bishopCheck = (targetSquare:number) => {
+
+const tempArray = []
+const startSquare = targetSquare
+let currentSquare =targetSquare
+
+if (columnFinder(currentSquare) !== firstColumn && rowFinder(currentSquare) !== lastRow){
+    for (let i=0; i< 8; i++){
```
Comment (AlexanderKolberg) [inline src/functions/moveChecks/bishopCheck.ts:11]: Could this be a reusable function?

File: src/functions/moveChecks/bishopCheck.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { diagonalRowNE, diagonalRowSW,rowFinder,column, firstColumn, columnFinder, row, lastColumn, lastRow, firstRow, diagonalRowSE, diagonalRowNW } from "../../global"
+import { allowedSquares } from "../../stores"
+
+export const bishopCheck = (targetSquare:number) => {
+
+const tempArray = []
+const startSquare = targetSquare
```
Comment (AlexanderKolberg) [inline src/functions/moveChecks/bishopCheck.ts:7]: Looks like you are not using this? Also, if you wanted to call it startSquare, you cold have done it in the arguments?

**Comment by valeriegc** [comment]
Merging as of now, will return to the logic improvements for some pieces at a later point of build.

**Pull request merged**